src/general.py

```diff
@@ -146,17 +259,32 @@ def generate_sub_object(sub_file: Path, test_name: str) -> htcondor2.Submit:
         job = htcondor2.Submit(f.read())
     if job is None:
         print(f"Error: Invalid submit file for test {test_name}")
-        print("Exiting...")
-        exit(1)
+        sys.exit(1)
 
     # add requirement to land on target ResourceName
+    req_expr = 'TARGET.GLIDEIN_ResourceName == "$(ResourceName)"'
     req = job.get("Requirements")
     if req is None:
-        job["Requirements"] = f'TARGET.GLIDEIN_ResourceName == "$(ResourceName)"'
+        job["Requirements"] = req_expr
     else:
-        job["Requirements"] = (
-            f'TARGET.GLIDEIN_ResourceName == "$(ResourceName)" && ' + req
-        )
+        job["Requirements"] = req_expr + f" && ({req})"
+
+    # add periodic removal statement
+    # job should be removed if it is in idle or running for more than 4 hours, if it ever
+    # goes on hold, or if it restarts more than 10 times
+    sec_in_4hr = 60 * 60 * 4
+    prdc_rm_expr = (
+        f"(JobStatus == 1 && CurrentTime-EnteredCurrentStatus > {sec_in_4hr}) || "
+        + f"(JobStatus == 2 && CurrentTime-EnteredCurrentStatus > {sec_in_4hr}) || "
+        + "(JobStatus == 5) || "
+        + "(NumShadowStarts > 10)"
+    )
+    prdc_rm = job.get("periodic_remove")
+    if prdc_rm is None:
+        job["periodic_remove"] = prdc_rm_expr
+    else:
+        job["periodic_remove"] = prdc_rm_expr + f" || ({prdc_rm})"
```

Comment:
Same idea as above if you want to make this a one line item:
`job["periodic_remove"] = prdc_rm_expr if prdc_rm is None else prdc_rm_expr + f" || ({prdc_rm})"